src/Surfnet/StepupMiddleware/CommandHandlingBundle/Tests/EventHandling/RecordEventsAndPublishToBusOnFirstCallEventListener.php

```diff
@@ -41,21 +41,38 @@ class OnFirstCallPublishesToBusAndCountingCallsEventListener implements EventLis
     private $toPublish;
 
     /**
-     * @param BufferedEventBus  $eventBus
+     * @var DomainMessage[]
+     */
+    private $recordedEvents = [];
+
+    /**
+     * @param BufferedEventBus $eventBus
      * @param DomainEventStream $toPublish
      */
     public function __construct(BufferedEventBus $eventBus, DomainEventStream $toPublish)
     {
-        $this->eventBus = $eventBus;
+        $this->eventBus  = $eventBus;
         $this->toPublish = $toPublish;
     }
 
+    /**
+     * @param DomainMessage $domainMessage
+     */
     public function handle(DomainMessage $domainMessage)
     {
+        $this->recordedEvents[] = $domainMessage;
+
         if ($this->callCount === 0) {
             $this->eventBus->publish($this->toPublish);
+            $this->callCount = 1;
```

Comment:
please make `callCount` something more relevant :wink: It seems to be used as boolean now, so perhaps make it a boolean (perhaps `firstEventHandled` or something?)